auth/api/iam/api.go

```diff
@@ -607,3 +619,16 @@ func (r Wrapper) accessTokenClientStore() storage.SessionStore {
 func (r Wrapper) accessTokenServerStore() storage.SessionStore {
 	return r.storageEngine.GetSessionDatabase().GetStore(accessTokenValidity, "serveraccesstoken")
 }
+
+// createOAuth2EndpointURL creates an OAuth2 endpoint URL for an owned did:web DID, for the given endpoint.
+// It creates a URL in the following format: https://<did:web host>/oauth2/<did>/<endpoint>
+func createOAuth2EndpointURL(webDID did.DID, endpointPath ...string) (*url.URL, error) {
+	didURL, err := didweb.DIDToURL(webDID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert DID to URL: %w", err)
+	}
+	didURL.RawPath = ""
```

Comment:
this deserves a comment

Reply:
Made it less fault prone by just creating a new `url.URL`.